src/gt4py/next/errors/excepthook.py

```diff
@@ -0,0 +1,77 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2023, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+"""
+Loading this module registers an excepthook that formats py:class:: DSLError.
+
+The excepthook is necessary because the default hook prints DSLErrors in an
+inconvenient way. The previously set excepthook is used to print all other
+errors.
+"""
+
+import os
+import sys
+from typing import Callable
+
+from . import exceptions, formatting
+
+
+def _get_verbose_exceptions_envvar() -> bool:
+    """Detect if the user enabled verbose exceptions in the environment variables."""
+    env_var_name = "GT4PY_VERBOSE_EXCEPTIONS"
+    if env_var_name in os.environ:
+        try:
+            return bool(os.environ[env_var_name])
```

Comment:
No no, that's bad, what's the easiest way to convert a string to bool?

Reply:
The easiest way to convert a string to a bool is exactly what you did, but as str is a sequence, the default conversion is to True if it is not empty.

I think what you want to do here is to explicitly compare the envvar content against "True", "TRUE", "1" or "ON" to set the flag to `True`, against "False", "FALSE", "0" or "OFF" to set the flag to `False` and raise a warning otherwise. There is no builtin shortcut in Python for this.

Reply:
Actually, I just remembered there is a super simple solution for this that I think we're using in some other places:
`bool(int(os.environ[env_var_name]))`
By converting the string to an int first, we guarantee that we follow meaningful semantics for "0" and "1" and any other value that is convertible to an int should be considered at least a warning.

Reply:
Unfortunately, this doesn't work:
```
>>> int("False")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: 'False'
```
I'll just use an explicit list.